### PR TITLE
Cognitive Services data-plane updates

### PR DESCRIPTION
**Pull request by milismsft**
Regen the Cognitive Services with an updated autorest.java autogen. 

Methods which used to take 15-20 arguments are now taking only the required parameters plus an object which wraps the optional parameters.

Added new "manager" classes which moves the authentication to the library vs the user's code.

Updated pom.xml to reflect the service team's expectation for /build.

### CODE REVIEW

File: azure-cognitiveservices/pom.xml [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    <tag>HEAD</tag>
+  </scm>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <legal><![CDATA[[INFO] Any downloads listed may be third party software.  Microsoft grants you no rights for third party software.]]></legal>
+    <testMode>playback</testMode>
+    <playbackServerPort>11080</playbackServerPort>
+    <azure.core.version>0.9.8</azure.core.version>
+    <azure.runtime.version>1.2.0</azure.runtime.version>
```
Comment (jianghaolu) [inline azure-cognitiveservices/pom.xml]: Bump this version?

File: azure-cognitiveservices/pom.xml [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+  </properties>
+
+  <developers>
+    <developer>
+      <id>microsoft</id>
+      <name>Microsoft</name>
+    </developer>
+  </developers>
+
+  <repositories>
```
Comment (jianghaolu) [inline azure-cognitiveservices/pom.xml]: Remove this section and `pluginRepositories` section below.

File: azure-cognitiveservices/search/bingvisualsearch/pom.xml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>com.microsoft.azure.cognitiveservices</groupId>
+    <artifactId>azure-cognitiveservices-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+    <artifactId>azure-cognitiveservices-visualsearch</artifactId>
+    <version>1.0.0-beta-SNAPSHOT</version>
```
Comment (jianghaolu) [inline azure-cognitiveservices/search/bingvisualsearch/pom.xml]: Why is this version different from every other pom?

File: azure-cognitiveservices/vision/contentmoderator/pom.xml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>com.microsoft.azure.cognitiveservices</groupId>
+    <artifactId>azure-cognitiveservices-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+    <artifactId>azure-cognitiveservices-contentmoderator</artifactId>
+    <version>1.0.0-beta-SNAPSHOT</version>
```
Comment (jianghaolu) [inline azure-cognitiveservices/vision/contentmoderator/pom.xml]: Same here

File: azure-cognitiveservices/vision/computervision/pom.xml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>com.microsoft.azure.cognitiveservices</groupId>
+    <artifactId>azure-cognitiveservices-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+    <artifactId>azure-cognitiveservices-computervision</artifactId>
+    <version>1.0.0-beta-SNAPSHOT</version>
```
Comment (jianghaolu) [inline azure-cognitiveservices/vision/computervision/pom.xml]: Same here

File: azure-cognitiveservices/pom.xml (hunk 1 above)
Comment (milismsft) [reply azure-cognitiveservices/pom.xml]: What's the version that I should use? 1.4.0?

File: azure-cognitiveservices/pom.xml (hunk 2 above)
Comment (milismsft) [reply azure-cognitiveservices/pom.xml]: Done

File: azure-cognitiveservices/search/bingvisualsearch/pom.xml (hunk 3 above)
Comment (milismsft) [reply azure-cognitiveservices/search/bingvisualsearch/pom.xml]: Bing Visual Search is still in "beta"... Kirthi's request.

File: azure-cognitiveservices/vision/contentmoderator/pom.xml (hunk 4 above)
Comment (milismsft) [reply azure-cognitiveservices/vision/contentmoderator/pom.xml]: Only Bing related services will go out without the "beta" (except Visual Search). The rest of the libraries will remain beta for now.

File: azure-cognitiveservices/vision/computervision/pom.xml (hunk 5 above)
Comment (milismsft) [reply azure-cognitiveservices/vision/computervision/pom.xml]: Only Bing related services will go out without the "beta" (except Visual Search). The rest of the libraries will remain beta for now.

**Pull request merged**